tests/test_dtypes.py

```diff
@@ -0,0 +1,157 @@
+    BusinessDay,
+    CustomBusinessDay,
+    Day,
+)
+
+
+def test_datetimetz_dtype() -> None:
+    dttz_dt = pd.DatetimeTZDtype(unit="ns", tz="UTC")
+    check(assert_type(dttz_dt, pd.DatetimeTZDtype), pd.DatetimeTZDtype)
+    check(
+        assert_type(
+            pd.DatetimeTZDtype(unit="ns", tz=timezone(timedelta(hours=1))),
+            pd.DatetimeTZDtype,
+        ),
+        pd.DatetimeTZDtype,
+    )
+    check(assert_type(dttz_dt.unit, Literal["ns"]), str)
+    check(assert_type(dttz_dt.tz, dt.tzinfo), dt.tzinfo)
+    check(assert_type(dttz_dt.name, str), str)
+    check(assert_type(dttz_dt.na_value, NaTType), NaTType)
+
+
+def test_period_dtype() -> None:
+    p_dt = pd.PeriodDtype(freq="D")
+    check(assert_type(p_dt, pd.PeriodDtype), pd.PeriodDtype)
+    check(assert_type(pd.PeriodDtype(freq=Day()), pd.PeriodDtype), pd.PeriodDtype)
+    check(
+        assert_type(pd.PeriodDtype(freq=BusinessDay()), pd.PeriodDtype), pd.PeriodDtype
+    )
+    check(
+        assert_type(pd.PeriodDtype(freq=CustomBusinessDay()), pd.PeriodDtype),
+        pd.PeriodDtype,
+    )
+    check(
+        assert_type(p_dt.freq, pd.tseries.offsets.BaseOffset),
+        pd.tseries.offsets.DateOffset,
+    )
+    check(assert_type(p_dt.na_value, NaTType), NaTType)
+    check(assert_type(p_dt.name, str), str)
+
+
+def test_interval_dtype() -> None:
+    i_dt = pd.IntervalDtype("int64")
+    check(assert_type(i_dt, pd.IntervalDtype), pd.IntervalDtype)
+    check(assert_type(pd.IntervalDtype(np.int64), pd.IntervalDtype), pd.IntervalDtype)
+    check(assert_type(pd.IntervalDtype(float), pd.IntervalDtype), pd.IntervalDtype)
+    check(assert_type(pd.IntervalDtype(complex), pd.IntervalDtype), pd.IntervalDtype)
+    check(
+        assert_type(pd.IntervalDtype(np.timedelta64), pd.IntervalDtype),
+        pd.IntervalDtype,
+    )
+    check(
+        assert_type(pd.IntervalDtype(np.datetime64), pd.IntervalDtype), pd.IntervalDtype
+    )
+
+
+def test_int64_dtype() -> None:
+    check(assert_type(pd.Int8Dtype(), pd.Int8Dtype), pd.Int8Dtype)
+    check(assert_type(pd.Int16Dtype(), pd.Int16Dtype), pd.Int16Dtype)
+    check(assert_type(pd.Int32Dtype(), pd.Int32Dtype), pd.Int32Dtype)
+    check(assert_type(pd.Int64Dtype(), pd.Int64Dtype), pd.Int64Dtype)
+    check(assert_type(pd.UInt8Dtype(), pd.UInt8Dtype), pd.UInt8Dtype)
+    check(assert_type(pd.UInt16Dtype(), pd.UInt16Dtype), pd.UInt16Dtype)
+    check(assert_type(pd.UInt32Dtype(), pd.UInt32Dtype), pd.UInt32Dtype)
+    check(assert_type(pd.UInt64Dtype(), pd.UInt64Dtype), pd.UInt64Dtype)
+
+    i64dt = pd.Int64Dtype()
+    check(assert_type(i64dt.itemsize, int), int)
+    check(assert_type(i64dt.na_value, NAType), NAType)
+    check(assert_type(i64dt.is_signed_integer, bool), bool)
+    check(assert_type(i64dt.is_unsigned_integer, bool), bool)
+    check(assert_type(i64dt.numpy_dtype, np.dtype), np.dtype)
+    check(assert_type(i64dt.construct_array_type(), "type[IntegerArray]"), type)
```

Comment:
with none of the above being documented, we can delete the tests

Reply:
Removed all but `construct_array_type` which is documented for the parent class.